### PR TITLE
dev-java/annogen: EAPI 7, move to qdox:1.12 | dev-java/qdox: remove unused slot 1.6

### PR DESCRIPTION
**Pull request by vaukai**

### CODE REVIEW

**Comment by vaukai** [comment]
@fordfrog 
I've played around with `mvn` and the `qdox` ebuild and found we could probably get rid of `dev-java/javacup` and `dev-java/jflex` if could translate the following into a command in the ebuild :

```
    195 [DEBUG] -----------------------------------------------------------------------
    196 [DEBUG] Goal:          org.apache.maven.plugins:maven-antrun-plugin:1.3:run (generate-sources)
    197 [DEBUG] Style:         Regular
    198 [DEBUG] Configuration: <?xml version="1.0" encoding="UTF-8"?>
    199 <configuration>
    200   <pluginArtifacts>${plugin.artifacts}</pluginArtifacts>
    201   <project>${project}</project>
    202   <sourceRoot>/var/tmp/portage/dev-java/qdox-1.12.1-r2/work/qdox-qdox-1.12.1/target/generated-sources/parser</sourceRoot>
    203   <tasks>
    204     <echo>OS Arch = amd64</echo>
    205     <exec failonerror="true" dir="/var/tmp/portage/dev-java/qdox-1.12.1-r2/work/qdox-qdox-1.12.1/target/generated-sources/parser/com/thoughtworks/qdox/parser/impl" executable="/var/tmp/portage/dev-java/qdox-1.12.1-r2/work/qdox-qdox-1.12.1/bootstrap/yacc.linux.x86_64">
    206       <arg value="-v"/>
    207       <arg value="-Jnorun"/>
    208       <arg value="-Jnoconstruct"/>
    209       <arg value="-Jclass=Parser"/>
    210       <arg value="-Jsemantic=Value"/>
    211       <arg value="-Jpackage=com.thoughtworks.qdox.parser.impl"/>
    212       <arg value="/var/tmp/portage/dev-java/qdox-1.12.1-r2/work/qdox-qdox-1.12.1/src/grammar/parser.y"/>
    213     </exec>
    214     <move file="/var/tmp/portage/dev-java/qdox-1.12.1-r2/work/qdox-qdox-1.12.1/target/generated-sources/parser/com/thoughtworks/qdox/parser/impl/y" tofile="/var/tmp/portage/dev-java/qdox-1.12.1-r2/work/qdox-qdox-1.12.1/target/yacc-states.txt" failonerror="false"/>${tasks}</tasks>
    215   <testSourceRoot>${testSourceRoot}</testSourceRoot>
    216 </configuration>

```

File: dev-java/annogen/annogen-0.1.0-r4.ebuild [hunk 1]
```diff
@@ -12,19 +12,15 @@ SRC_URI="http://dist.codehaus.org/${PN}/distributions/${P}.zip"
...
-CDEPEND="
-	dev-java/qdox:1.6
-	dev-java/ant-core:0"
-
-RDEPEND="
-	${CDEPEND}
+CDEPEND="dev-java/ant-core:0
+	dev-java/qdox:1.12"
+RDEPEND="${CDEPEND}
 	>=virtual/jre-1.6"
```
Comment (fordfrog) [inline dev-java/annogen/annogen-0.1.0-r4.ebuild:20]: should be lifted up to 1.8

File: dev-java/annogen/annogen-0.1.0-r4.ebuild [hunk 2]
```diff
@@ -12,19 +12,15 @@ SRC_URI="http://dist.codehaus.org/${PN}/distributions/${P}.zip"
...
-
-DEPEND="
-	${CDEPEND}
+DEPEND="${CDEPEND}
 	>=virtual/jdk-1.6"
```
Comment (fordfrog) [inline dev-java/annogen/annogen-0.1.0-r4.ebuild:22]: should be lifted up to 1.8

File: dev-java/annogen/annogen-0.1.0-r4.ebuild [hunk 3]
```diff
@@ -12,19 +12,15 @@ SRC_URI="http://dist.codehaus.org/${PN}/distributions/${P}.zip"
 
 LICENSE="BSD"
 SLOT="0"
-KEYWORDS="amd64 ~ppc64 ~x86"
+KEYWORDS="~amd64 ~ppc64 ~x86"
```
Comment (fordfrog) [inline dev-java/annogen/annogen-0.1.0-r4.ebuild:15]: here you're dropping the package to unstable and later you remove the stable version. keep `amd64` stable in this ebuild and if my tests pass, i will commit it straight to stable.

**Pull request closed without merging**